### PR TITLE
Fixing #84 and other issues, added more tests

### PR DESCRIPTION
**Pull request by mpenkov**

### CODE REVIEW

**Comment by tmylk** [comment]
Thanks. could you add a description of the fix to the changelog?

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,10 @@
+* 1.3.5, 6th August 2015
+
+  - Move gzipstream module to smart_open package
+  - Make sure reader objects never return None
+  - Fix issue #85
```
Comment (piskvorky) [inline CHANGELOG.rst]: Too cryptic -- describe the actual change.

Also, we can hyperlink to the actual PR / more information, where appropriate (not just text).

File: CHANGELOG.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@
   - Ensure reader objects never return None
   - Ensure read functions never return more bytes than asked for
   - Add support for reading gzipped objects until EOF, e.g. read()
-  - Fix issue #84
+  - Add missing parameter to read_from_buffer call (https://github.com/RaRe-Technologies/smart_open/issues/84)
```
Comment (piskvorky) [inline CHANGELOG.rst]: @tmylk let's change CHANGELOG from `.rst` to `.md`, so it's easier for people to hyperlink. (the markdown syntax is a little simpler to remember)

**Comment by tmylk** [comment]
Thanks a lot for the great PR! Squashed the CHANGELOG commits and merged in https://github.com/RaRe-Technologies/smart_open/commit/9cdcd1758b2f2f9fa190f58e7744c34593a968f1

**Pull request closed without merging**